### ISSUE TITLE
Blood pressure indicate

### ISSUE THREAD

**Issue by MHrezdovic** [randdusing/cordova-plugin-bluetoothle#132]
Hi guys, 

i have a issue i would like to read data from blood pressure meter but somehow im unable to to get anything from the BP meter. After subscribing to BP monitor i always get 0 or error. Someone have an ide what is the life cycle of BP measurement. e.g.: Discover connect subscribe read disconect close.

Or if someone made some similar app can you provide the source code?

thx

**Comment by randdusing** [comment]
What device, services and characteristics are you using?

**Comment by MHrezdovic** [comment]
I have  Medisana BU 550 Connect it has Blood pressure service 1810 and
blood pressure characterestic 2A35. From what i learned the Blood pressure
measurement should use indicate so the client(smartphone) should somehow
respond with ack to the server(BPM) but after i subscribe the client to BP
service and characteristic i get nothing.

2015-02-03 3:31 GMT+01:00 Rand Dusing <notifications@github.com>:

> What device, services and characteristics are you using?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/randdusing/BluetoothLE/issues/132#issuecomment-72581464>
> .
>



-- 
S pozdravom
Michal Hrezdovic

**Comment by randdusing** [comment]
Are you getting any returned data value? And how are you parsing it?

**Comment by MHrezdovic** [comment]
Im not getting anything i somehow cant figgure out how. I tryed to read the
characteristic but they seems to expect some command to start transmiting
the data

2015-02-03 20:38 GMT+01:00 Rand Dusing <notifications@github.com>:

> Are you getting any returned data value? And how are you parsing it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/randdusing/BluetoothLE/issues/132#issuecomment-72718952>
> .
>



-- 
S pozdravom
Michal Hrezdovic

**Comment by randdusing** [comment]
Check out the example about subscribing to a heart rate monitor. Blood pressure appears similar and uses the client configuration characteristic. Main difference seems to be notify vs indication.

https://developer.bluetooth.org/gatt/services/Pages/ServiceViewer.aspx?u=org.bluetooth.service.blood_pressure.xml

**Comment by MHrezdovic** [comment]
Yes, i figure that out but i am clueless about how the indication works. I
believe that on indicate client needs to send some message or some
acknowlidgement to the server. But cant find anywhere what this ack should
be.

2015-02-04 11:13 GMT+01:00 Rand Dusing <notifications@github.com>:

> Check out the example about subscribing to a heart rate monitor. Blood
> pressure appears similar and uses the client configuration characteristic.
> Main difference seems to be notify vs indication.
>
>
> https://developer.bluetooth.org/gatt/services/Pages/ServiceViewer.aspx?u=org.bluetooth.service.blood_pressure.xml
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/randdusing/BluetoothLE/issues/132#issuecomment-72830245>
> .
>



-- 
S pozdravom
Michal Hrezdovic

**Comment by randdusing** [comment]
Ahh, I see what you mean. I don't see anything in the API about acknowledging indications, so I assumed it was something the Bluetooth stack handled.

**Comment by MHrezdovic** [comment]
Well you might be right but i did not come accross some file that would
shed some light into this topic :(

2015-02-04 11:37 GMT+01:00 Rand Dusing <notifications@github.com>:

> Ahh, I see what you mean. I don't see anything in the API about
> acknowledging indications, so I assumed it was something the Bluetooth
> stack handled.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/randdusing/BluetoothLE/issues/132#issuecomment-72833467>
> .
>



-- 
S pozdravom
Michal Hrezdovic

**Comment by randdusing** [comment]
Were you able to resolve your issue?

**Comment by MHrezdovic** [comment]
Hi,

No im somehow unable to acknowlidge the transmition so now luck so far...

2015-02-27 1:13 GMT+01:00 Rand Dusing <notifications@github.com>:

> Were you able to resolve your issue?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/randdusing/BluetoothLE/issues/132#issuecomment-76304080>
> .
>



-- 
S pozdravom
Michal Hrezdovic

**Comment by randdusing** [comment]
Still having issues? I wouldn't mind testing it on a physical blood pressure device, but it looks like the Medisanas are pretty expensive to obtain in the US. Do you know of any other brands?

**Comment by MHrezdovic** [comment]
yup, but i didnt gave it much time afterwards...never managed to
acknowlidge the transmition

2016-01-25 0:17 GMT+01:00 Rand Dusing <notifications@github.com>:

> Still having issues? I wouldn't mind testing it on a physical blood
> pressure device, but it looks like the Medisanas are pretty expensive to
> obtain in the US. Do you know of any other brands?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/randdusing/BluetoothLE/issues/132#issuecomment-174351488>
> .
>



-- 
S pozdravom
Michal Hrezdovic

**Comment by randdusing** [comment]
I tested this out with the server side code I wrote. I created a characteristic with the notify property and one with the indicate property. Using another iOS device, I was able to successfully subscribe to both characteristics.

On Android, I could subscribe to the characteristic with notify, but not with indicate. I then tried again by passing isNotification => false within the subscribe params, and it worked. By the way, I'm changing subscribe to just automatically detect whether the notify or indicate property is set, so that the app doesn't need to specify isNotification anymore.

So with the change that will be committed to the master branch within an hour or two, there's nothing that needs to be done at the app level to handle indications. Let me know if this doesn't work.
